apps/framework-cli/src/infrastructure/olap/clickhouse/queries.rs

```diff
@@ -129,30 +173,378 @@ impl<'a> TryFrom<&'a str> for ClickhouseEngine {
+            }
+        }
+
+        format!("{:x}", hasher.finish())
+    }
+}
+
 pub fn create_table_query(
     db_name: &str,
     table: ClickHouseTable,
 ) -> Result<String, ClickhouseError> {
     let mut reg = Handlebars::new();
     reg.register_escape_fn(no_escape);
 
-    let engine = match table.engine {
-        ClickhouseEngine::MergeTree => "MergeTree",
+    let engine = match &table.engine {
+        ClickhouseEngine::MergeTree => "MergeTree".to_string(),
         ClickhouseEngine::ReplacingMergeTree => {
             if table.order_by.is_empty() {
                 return Err(ClickhouseError::InvalidParameters {
                     message: "ReplacingMergeTree requires an order by clause".to_string(),
                 });
             }
-            "ReplacingMergeTree"
+            "ReplacingMergeTree".to_string()
+        }
+        ClickhouseEngine::AggregatingMergeTree => "AggregatingMergeTree".to_string(),
+        ClickhouseEngine::SummingMergeTree => "SummingMergeTree".to_string(),
+        ClickhouseEngine::S3Queue {
+            s3_path,
+            format,
+            compression,
+            headers: _headers, // TODO: Handle headers in future if needed
+            aws_access_key_id,
+            aws_secret_access_key,
+            ..
+        } => {
+            // Build the engine string based on available parameters
+            let mut engine_parts = vec![format!("'{}'", s3_path)];
+
+            // Handle credentials from the engine configuration
+            if let (Some(key_id), Some(secret)) = (aws_access_key_id, aws_secret_access_key) {
+                engine_parts.push(format!("'{}'", key_id));
+                engine_parts.push(format!("'{}'", secret));
+            } else {
+                // Default to NOSIGN for public buckets or when credentials are not available
+                engine_parts.push("'NOSIGN'".to_string());
+            }
+
+            engine_parts.push(format!("'{}'", format));
+
+            // Add compression if specified
+            if let Some(comp) = compression {
+                engine_parts.push(format!("'{}'", comp));
+            }
+
+            format!("S3Queue({})", engine_parts.join(", "))
         }
-        ClickhouseEngine::AggregatingMergeTree => "AggregatingMergeTree",
-        ClickhouseEngine::SummingMergeTree => "SummingMergeTree",
+    };
+
+    // Format settings from table.table_settings
+    let settings = if let Some(ref table_settings) = table.table_settings {
+        if !table_settings.is_empty() {
+            let mut settings_pairs: Vec<(String, String)> = table_settings
+                .iter()
+                .map(|(key, value)| (key.clone(), value.clone()))
+                .collect();
+            settings_pairs.sort_by(|a, b| a.0.cmp(&b.0)); // Sort by key for deterministic order
+            let settings_strs: Vec<String> = settings_pairs
+                .iter()
+                .map(|(key, value)| format!("{} = '{}'", key, value))
```

Comment:
I wonder if need to detect what the value type is. Not entirely sure how clickhouse accepts numbers/booleans here.